### ISSUE TITLE
cube.convert_units converting from 'unknown'

### ISSUE THREAD

**Issue by rcomer** [SciTools/iris#1977]
I just noticed that `cube.convert_units` works when the cube doesn't have an existing unit.  E.g.:

``` python
mycube = iris.cube.Cube(1.)
mycube.convert_units('mm day-1')
print mycube
print mycube.data
```

produces this output:

```
unknown / (mm day-1) 
1.0
```

Shouldn't `convert_units` produce an exception if it doesn't have a unit to map from?

**Comment by ajdawson** [comment]
I agree. The code for `convert_units` looks like this:

```python
def convert_units(self, unit):
    # If the cube has units convert the data.
    if not self.units.is_unknown():
        self.data = self.units.convert(self.data, unit)
    self.units = unit
```

So we are happily changing the unit to the new unit without manipulating data values. In my mind this corresponds to just setting the `units` attribute. Therefore, we should raise an exception when converting units from `'unknown'` and just instruct the user to set the units attribute directly, since this operation is directly equivalent to the current (surprising) behaviour of `convert_units`:

```python
def convert_units(self, unit):
    if self.units.is_unknown():
        raise SomeRelevantException('cannot convert from unknown units to {!s}, '
                                    'set the "units" attribute instead'.format(unit))
    self.data = self.units.convert(self.data, unit)
    self.units = unit
```

**Comment by rcomer** [comment]
Fixed at #2700.
